File: python/1-9/problem4.py
def palindrome(n):
    '''Returns a boolean on weather a number is a palindrome.'''
    return n == int(str(n)[::-1])


def palindrome_product(start, end):
    '''Returns a list of palindrome products.'''
    r = range(start, end)
    return [x * y for x in r for y in r if palindrome(x * y)]


if __name__ == '__main__':
    print(max(palindrome_product(100, 1000)))

File: js/1-9/problem4.js
const prototypes = require('../prototypes');

// Returns an array of all the palindrome numbers between 'start' and 'end'.
var palindrome = function(start, end) {
	var answer = [];
	var reverseNum = n => parseInt(n.toString().split('').reverse().join(''));

	for (let i = start; i < end; i++) {
		for (let j = start; j < end; j++) {
			var product = i * j
			if (product === reverseNum(product)) {
				answer.push(product);
			}
		}
	}
	return answer;
};

console.log(palindrome(100, 1000).max());
